central/reprocessor/reprocessor.go

```diff
@@ -460,9 +462,28 @@ func (l *loopImpl) reprocessNodes() {
 }
 
 func (l *loopImpl) reprocessWatchedImage(name string) bool {
-	img, err := imageEnricher.EnrichImageByName(emptyCtx, l.imageEnricher, imageEnricher.EnrichmentContext{
+	enrichmentCtx := imageEnricher.EnrichmentContext{
 		FetchOpt: imageEnricher.IgnoreExistingImages,
-	}, name)
+	}
+
+	ctx := emptyCtx
+	if features.DelegateWatchedImageReprocessing.Enabled() {
+		// Delegated scanning performs an access check, an empty context will
+		// trigger a hard panic. If an image watch was successfully created
+		// we can assume at that time the user had access, so we create a
+		// context that has access to delegate the scan.
+		ctx = sac.WithGlobalAccessScopeChecker(
```

Comment:
I'm definitely not an expert for our SAC stuff, but it looks like this gives our background context access to read all images? Should we just make this a global var so we don't have to recreate this every time?

Reply:
+1 for creating a var like "readImagesCtx" or something along the lines, note that this will create a _new_ context though and not give `context.Background()` access.

Reply:
great suggestion, done: https://github.com/stackrox/stackrox/pull/11121/commits/6d1869c6592f8a16157135e8d18dac7a39f368f0